specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-01-10/service.json

```diff
@@ -16176,6 +16176,10 @@
             "name": "DiskAccountType",
             "modelAsString": true
           }
+        },
+        "diskEncryptionSetId": {
```

Comment:
This appears to be the only non-readonly field your adding which constitutes a breaking change and requires a new api version.

See 'New property added to response'
https://microsoft.sharepoint.com/:w:/t/azureresourcemanagerteam/EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw?e=xEcppo

I see you have the response with this field marked as readonly but it still breaks the get-put scenario called out in the doc.

Reply:
As this is an optional input property and which cannot be updated once set, won’t this be safe in the PUT -> GET -> PUT scenario? From service side, we retain the existing value of this property if a new PUT comes where this value is not set (the property is tagged as “read-only” in the response).

Reply:
There is [x-ms-mutability](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-mutability) which might help in this situation (Though I am not sure how different code generator would handle it. Well at least from reader the intention would be clearer).

I guess we know many service actually use merge in PUT to handle backward compatibility. But that is really a service by service (or even endpoint by endpoint?) solution, so I think ARM had to ask.